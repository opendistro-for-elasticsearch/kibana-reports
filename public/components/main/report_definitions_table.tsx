/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import {
  EuiLink,
  EuiInMemoryTable,
  EuiButton,
  EuiEmptyPrompt,
  EuiText,
} from '@elastic/eui';
import { report_definitions } from './test_data';

export const scheduled_report_columns = [
  {
    field: 'reportName',
    name: 'Name',
    sortable: true,
    truncateText: true,
  },
  {
    field: 'type',
    name: 'Type',
    sortable: true,
    truncateText: false,
  },
  {
    field: 'owner',
    name: 'Owner',
    sortable: true,
    truncateText: false,
  },
  {
    field: 'source',
    name: 'Source',
    render: (username: string) => (
      <EuiLink href={'#'} target="_blank">
        {username}
      </EuiLink>
    ),
  },
  {
    field: 'lastUpdated',
    name: 'Last Updated',
    truncateText: true,
  },
  {
    field: 'details',
    name: 'Details',
    sortable: false,
    truncateText: true,
  },
  {
    field: 'status',
    name: 'Status',
    sortable: true,
    truncateText: false,
  },
];

const empty_message_report_definitions = (
  <EuiEmptyPrompt
    title={<h3>No report definitions to display</h3>}
    titleSize="xs"
    body={
      <div>
        <EuiText>Create a new report definition to get started</EuiText>
        <EuiText>
          To learn more, see{' '}
          <EuiLink>Get started with Kibana reporting</EuiLink>
        </EuiText>
      </div>
    }
    actions={
      <div>
        <EuiButton
          onClick={() => {
            window.location.assign('opendistro-kibana-reports#/create');
          }}
        >
          Create
        </EuiButton>
      </div>
    }
  />
);

const report_definitions_search = {
  box: {
    incremental: true,
  },
  filters: [],
};

export function ReportDefinitions(props) {
  const { pagination, getRowProps } = props;
  return (
    <div>
      <EuiInMemoryTable
        items={report_definitions}
        itemId="id"
        loading={false}
        message={empty_message_report_definitions}
        columns={scheduled_report_columns}
        search={report_definitions_search}
        pagination={pagination}
        sorting={true}
        isSelectable={true}
        rowProps={getRowProps}
      />
    </div>
  );
}
