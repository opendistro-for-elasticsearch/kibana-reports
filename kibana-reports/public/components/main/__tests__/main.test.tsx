/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Main } from '../main';
import httpClientMock from '../../../../test/httpMockClient';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { act } from 'react-dom/test-utils';

function setBreadcrumbs(array: []) {
  jest.fn();
}

describe('<Main /> panel', () => {
  configure({ adapter: new Adapter() });
  test('render component', async (done) => {
    window = Object.create(window);
    Object.defineProperty(window, 'location', {
      configurable: true,
      value: {
        assign: jest.fn(),
        href: 'opendistro_kibana_reports#/',
      },
    });

    const { container } = await render(
      <Main httpClient={httpClientMock} setBreadcrumbs={setBreadcrumbs} />
    );

    expect(container.firstChild).toMatchSnapshot();
    done();
  });

  test('render component after create success', async () => {
    delete window.location;

    Object.defineProperty(window, 'location', {
      configurable: true,
      value: {
        assign: jest.fn(),
        href: 'opendistro_kibana_reports#/create=success',
      },
    });

    const { container } = render(
      <Main httpClient={httpClientMock} setBreadcrumbs={setBreadcrumbs} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('render component after edit success', async () => {
    delete window.location;

    Object.defineProperty(window, 'location', {
      configurable: true,
      value: {
        assign: jest.fn(),
        href: 'opendistro_kibana_reports#/edit=success',
      },
    });

    const { container } = render(
      <Main httpClient={httpClientMock} setBreadcrumbs={setBreadcrumbs} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('test refresh reports definitions button', async () => {
    const promise = Promise.resolve();
    const data = [
      {
        _id: 'abcdefg',
        _source: {
          query_url: '/app/visualize/edit/1234567890',
          state: 'Created',
          time_created: 123456789,
          time_from: 123456789,
          time_to: 1234567890,
          report_definition: {
            report_params: {
              report_name: 'test create report definition trigger',
              report_source: 'Dashboard',
              description: '',
              core_params: {
                base_url: 'http://localhost:5601',
                report_format: 'png',
                header: '',
                footer: '',
                time_duration: 'PT30M',
              },
            },
            delivery: {
              delivery_type: '',
              delivery_params: {},
            },
            trigger: {
              trigger_type: 'Schedule',
              trigger_params: {},
            },
          },
        },
      },
    ];

    httpClientMock.get = jest.fn().mockResolvedValue({
      data,
    });

    const component = mount(
      <Main httpClient={httpClientMock} setBreadcrumbs={setBreadcrumbs} />
    );
    await act(() => promise);

    const generate = component.find('button').at(7);
    generate.simulate('click');
    await act(() => promise);
  });

  test('test refresh reports table button', async () => {
    const promise = Promise.resolve();
    const data = [
      {
        _id: 'abcdefg',
        _source: {
          query_url: '/app/visualize/edit/1234567890',
          state: 'Created',
          time_created: 123456789,
          time_from: 123456789,
          time_to: 1234567890,
          report_definition: {
            report_params: {
              report_name: 'test create report definition trigger',
              report_source: 'Dashboard',
              description: '',
              core_params: {
                base_url: 'http://localhost:5601',
                report_format: 'png',
                header: '',
                footer: '',
                time_duration: 'PT30M',
              },
            },
            delivery: {
              delivery_type: '',
              delivery_params: {},
            },
            trigger: {
              trigger_type: 'Schedule',
              trigger_params: {},
            },
          },
        },
      },
    ];

    httpClientMock.get = jest.fn().mockResolvedValue({
      data,
    });

    const component = mount(
      <Main httpClient={httpClientMock} setBreadcrumbs={setBreadcrumbs} />
    );
    await act(() => promise);

    const generate = component.find('button').at(0);
    generate.simulate('click');
    await act(() => promise);
  });

  // TODO: mock catch() error response to contain status code 
  test.skip('test error toasts posted', async () => {
    jest.spyOn(console, 'log').mockImplementation(() => {}); // silence console log error from main
    const promise = Promise.resolve();

    httpClientMock.get = jest.fn().mockResolvedValue({
      response: null,
    });

    const component = mount(
      <Main httpClient={httpClientMock} setBreadcrumbs={setBreadcrumbs} />
    );
    const generate = component.find('button').at(7);
    try {
      generate.simulate('click');
      await act(() => promise);
    } catch (e) {
      await act(() => promise);
    }
  });
});
