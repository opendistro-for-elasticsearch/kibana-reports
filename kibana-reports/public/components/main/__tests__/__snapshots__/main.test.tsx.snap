// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> panel render component 1`] = `
<div>
  <div
    class="euiPanel euiPanel--paddingLarge"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceEvenly euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <h2
          class="euiTitle euiTitle--medium"
        >
          Reports
           
          <p
            style="color: gray; display: inline;"
          >
             (
            0
            )
          </p>
        </h2>
      </div>
      <span
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButton__content"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiIcon-isLoading euiButton__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            />
            <span
              class="euiButton__text"
            >
              Refresh
            </span>
          </span>
        </button>
      </span>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
    />
    <div>
      <div
        class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
      >
        <div
          class="euiFlexItem euiSearchBar__searchHolder"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
                class="euiFieldSearch euiFieldSearch--fullWidth"
                placeholder="Search by Report ID"
                type="search"
                value=""
              />
              <div
                class="euiFormControlLayoutIcons"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiBasicTable"
      >
        <div>
          <div
            class="euiTableHeaderMobile"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              />
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              />
            </div>
          </div>
          <table
            class="euiTable euiTable--responsive euiTable--auto"
            id="random_html_id"
            tabindex="-1"
          >
            <caption
              class="euiScreenReaderOnly euiTableCaption"
            />
            <thead>
              <tr>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_reportName_0"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Report ID"
                    >
                      Report ID
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_reportName_1"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Source"
                    >
                      Source
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_timeCreated_2"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Creation time"
                    >
                      Creation time
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_url_3"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Time period"
                    >
                      Time period
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_id_4"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Generate"
                    >
                      Generate
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="euiTableRow"
              >
                <td
                  class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                  colspan="5"
                >
                  <div
                    class="euiTableCellContent euiTableCellContent--alignCenter"
                  >
                    <span
                      class="euiTableCellContent__text"
                    >
                      <div
                        class="euiEmptyPrompt"
                      >
                        <span
                          class="euiTextColor euiTextColor--subdued"
                        >
                          <h3
                            class="euiTitle euiTitle--xsmall"
                          >
                            No reports to display
                          </h3>
                          <div
                            class="euiSpacer euiSpacer--m"
                          />
                          <div
                            class="euiText euiText--medium"
                          >
                            <div>
                              <div
                                class="euiText euiText--medium"
                              >
                                Create a report definition, or share/download a report from a dashboard, saved search or visualization.
                              </div>
                              <div
                                class="euiText euiText--medium"
                              >
                                To learn more, see
                                 
                                <button
                                  class="euiLink euiLink--primary"
                                  type="button"
                                >
                                  Get started with Kibana reporting 
                                  <svg
                                    aria-hidden="true"
                                    class="euiIcon euiIcon--medium euiIcon-isLoading"
                                    focusable="false"
                                    height="16"
                                    role="img"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </span>
                      </div>
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
    class="euiGlobalToastList euiGlobalToastList--right"
    role="region"
  />
</div>
`;

exports[`<Main /> panel render component after create success 1`] = `
<div>
  <div
    class="euiPanel euiPanel--paddingLarge"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceEvenly euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <h2
          class="euiTitle euiTitle--medium"
        >
          Reports
           
          <p
            style="color: gray; display: inline;"
          >
             (
            0
            )
          </p>
        </h2>
      </div>
      <span
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButton__content"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiButton__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.228 2.942a.5.5 0 11-.538.842A5 5 0 1013 8a.5.5 0 111 0 6 6 0 11-2.772-5.058zM14 1.5v3A1.5 1.5 0 0112.5 6h-3a.5.5 0 010-1h3a.5.5 0 00.5-.5v-3a.5.5 0 111 0z"
              />
            </svg>
            <span
              class="euiButton__text"
            >
              Refresh
            </span>
          </span>
        </button>
      </span>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
    />
    <div>
      <div
        class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
      >
        <div
          class="euiFlexItem euiSearchBar__searchHolder"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
                class="euiFieldSearch euiFieldSearch--fullWidth"
                placeholder="Search by Report ID"
                type="search"
                value=""
              />
              <div
                class="euiFormControlLayoutIcons"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.271 11.978l3.872 3.873a.502.502 0 00.708 0 .502.502 0 000-.708l-3.565-3.564c2.38-2.747 2.267-6.923-.342-9.532-2.73-2.73-7.17-2.73-9.898 0-2.728 2.729-2.728 7.17 0 9.9a6.955 6.955 0 004.949 2.05.5.5 0 000-1 5.96 5.96 0 01-4.242-1.757 6.01 6.01 0 010-8.486c2.337-2.34 6.143-2.34 8.484 0a6.01 6.01 0 010 8.486.5.5 0 00.034.738z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiBasicTable"
      >
        <div>
          <div
            class="euiTableHeaderMobile"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              />
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              />
            </div>
          </div>
          <table
            class="euiTable euiTable--responsive euiTable--auto"
            id="random_html_id"
            tabindex="-1"
          >
            <caption
              class="euiScreenReaderOnly euiTableCaption"
            />
            <thead>
              <tr>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_reportName_0"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Report ID"
                    >
                      Report ID
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_reportName_1"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Source"
                    >
                      Source
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_timeCreated_2"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Creation time"
                    >
                      Creation time
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_url_3"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Time period"
                    >
                      Time period
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_id_4"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Generate"
                    >
                      Generate
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="euiTableRow"
              >
                <td
                  class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                  colspan="5"
                >
                  <div
                    class="euiTableCellContent euiTableCellContent--alignCenter"
                  >
                    <span
                      class="euiTableCellContent__text"
                    >
                      <div
                        class="euiEmptyPrompt"
                      >
                        <span
                          class="euiTextColor euiTextColor--subdued"
                        >
                          <h3
                            class="euiTitle euiTitle--xsmall"
                          >
                            No reports to display
                          </h3>
                          <div
                            class="euiSpacer euiSpacer--m"
                          />
                          <div
                            class="euiText euiText--medium"
                          >
                            <div>
                              <div
                                class="euiText euiText--medium"
                              >
                                Create a report definition, or share/download a report from a dashboard, saved search or visualization.
                              </div>
                              <div
                                class="euiText euiText--medium"
                              >
                                To learn more, see
                                 
                                <button
                                  class="euiLink euiLink--primary"
                                  type="button"
                                >
                                  Get started with Kibana reporting 
                                  <svg
                                    aria-hidden="true"
                                    class="euiIcon euiIcon--medium"
                                    focusable="false"
                                    height="16"
                                    role="img"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M13 8.5a.5.5 0 111 0V12a2 2 0 01-2 2H4a2 2 0 01-2-2V4a2 2 0 012-2h3.5a.5.5 0 010 1H4a1 1 0 00-1 1v8a1 1 0 001 1h8a1 1 0 001-1V8.5zm-5.12.339a.5.5 0 11-.706-.707L13.305 2H10.5a.5.5 0 110-1H14a1 1 0 011 1v3.5a.5.5 0 11-1 0V2.72L7.88 8.838z"
                                    />
                                  </svg>
                                </button>
                              </div>
                            </div>
                          </div>
                        </span>
                      </div>
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
    class="euiGlobalToastList euiGlobalToastList--right"
    role="region"
  >
    <div
      class="euiToast euiToast--success euiGlobalToastListItem"
      id="createReportDefinitionSuccessToast"
    >
      <p
        class="euiScreenReaderOnly"
      >
        A new notification appears
      </p>
      <div
        aria-label="Notification"
        class="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiToastHeader__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
        <span
          class="euiToastHeader__title"
        >
          Successfully created report definition.
        </span>
      </div>
      <button
        aria-label="Dismiss toast"
        class="euiToast__closeButton"
        data-test-subj="toastCloseButton"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<Main /> panel render component after edit success 1`] = `
<div>
  <div
    class="euiPanel euiPanel--paddingLarge"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceEvenly euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <h2
          class="euiTitle euiTitle--medium"
        >
          Reports
           
          <p
            style="color: gray; display: inline;"
          >
             (
            0
            )
          </p>
        </h2>
      </div>
      <span
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButton__content"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiButton__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.228 2.942a.5.5 0 11-.538.842A5 5 0 1013 8a.5.5 0 111 0 6 6 0 11-2.772-5.058zM14 1.5v3A1.5 1.5 0 0112.5 6h-3a.5.5 0 010-1h3a.5.5 0 00.5-.5v-3a.5.5 0 111 0z"
              />
            </svg>
            <span
              class="euiButton__text"
            >
              Refresh
            </span>
          </span>
        </button>
      </span>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
    />
    <div>
      <div
        class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
      >
        <div
          class="euiFlexItem euiSearchBar__searchHolder"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
                class="euiFieldSearch euiFieldSearch--fullWidth"
                placeholder="Search by Report ID"
                type="search"
                value=""
              />
              <div
                class="euiFormControlLayoutIcons"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.271 11.978l3.872 3.873a.502.502 0 00.708 0 .502.502 0 000-.708l-3.565-3.564c2.38-2.747 2.267-6.923-.342-9.532-2.73-2.73-7.17-2.73-9.898 0-2.728 2.729-2.728 7.17 0 9.9a6.955 6.955 0 004.949 2.05.5.5 0 000-1 5.96 5.96 0 01-4.242-1.757 6.01 6.01 0 010-8.486c2.337-2.34 6.143-2.34 8.484 0a6.01 6.01 0 010 8.486.5.5 0 00.034.738z"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiBasicTable"
      >
        <div>
          <div
            class="euiTableHeaderMobile"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              />
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              />
            </div>
          </div>
          <table
            class="euiTable euiTable--responsive euiTable--auto"
            id="random_html_id"
            tabindex="-1"
          >
            <caption
              class="euiScreenReaderOnly euiTableCaption"
            />
            <thead>
              <tr>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_reportName_0"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Report ID"
                    >
                      Report ID
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_reportName_1"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Source"
                    >
                      Source
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_timeCreated_2"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Creation time"
                    >
                      Creation time
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_url_3"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Time period"
                    >
                      Time period
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_id_4"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Generate"
                    >
                      Generate
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="euiTableRow"
              >
                <td
                  class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                  colspan="5"
                >
                  <div
                    class="euiTableCellContent euiTableCellContent--alignCenter"
                  >
                    <span
                      class="euiTableCellContent__text"
                    >
                      <div
                        class="euiEmptyPrompt"
                      >
                        <span
                          class="euiTextColor euiTextColor--subdued"
                        >
                          <h3
                            class="euiTitle euiTitle--xsmall"
                          >
                            No reports to display
                          </h3>
                          <div
                            class="euiSpacer euiSpacer--m"
                          />
                          <div
                            class="euiText euiText--medium"
                          >
                            <div>
                              <div
                                class="euiText euiText--medium"
                              >
                                Create a report definition, or share/download a report from a dashboard, saved search or visualization.
                              </div>
                              <div
                                class="euiText euiText--medium"
                              >
                                To learn more, see
                                 
                                <button
                                  class="euiLink euiLink--primary"
                                  type="button"
                                >
                                  Get started with Kibana reporting 
                                  <svg
                                    aria-hidden="true"
                                    class="euiIcon euiIcon--medium"
                                    focusable="false"
                                    height="16"
                                    role="img"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M13 8.5a.5.5 0 111 0V12a2 2 0 01-2 2H4a2 2 0 01-2-2V4a2 2 0 012-2h3.5a.5.5 0 010 1H4a1 1 0 00-1 1v8a1 1 0 001 1h8a1 1 0 001-1V8.5zm-5.12.339a.5.5 0 11-.706-.707L13.305 2H10.5a.5.5 0 110-1H14a1 1 0 011 1v3.5a.5.5 0 11-1 0V2.72L7.88 8.838z"
                                    />
                                  </svg>
                                </button>
                              </div>
                            </div>
                          </div>
                        </span>
                      </div>
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-live="polite"
    class="euiGlobalToastList euiGlobalToastList--right"
    role="region"
  >
    <div
      class="euiToast euiToast--success euiGlobalToastListItem"
      id="editReportDefinitionSuccessToast"
    >
      <p
        class="euiScreenReaderOnly"
      >
        A new notification appears
      </p>
      <div
        aria-label="Notification"
        class="euiToastHeader"
        data-test-subj="euiToastHeader"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiToastHeader__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.5 12a.502.502 0 01-.354-.146l-4-4a.502.502 0 01.708-.708L6.5 10.793l6.646-6.647a.502.502 0 01.708.708l-7 7A.502.502 0 016.5 12"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="euiToastHeader__title"
        >
          Successfully updated report definition.
        </span>
      </div>
      <button
        aria-label="Dismiss toast"
        class="euiToast__closeButton"
        data-test-subj="toastCloseButton"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.293 8L3.146 3.854a.5.5 0 11.708-.708L8 7.293l4.146-4.147a.5.5 0 01.708.708L8.707 8l4.147 4.146a.5.5 0 01-.708.708L8 8.707l-4.146 4.147a.5.5 0 01-.708-.708L7.293 8z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
